### PR TITLE
Fix softmax gradient.

### PR DESCRIPTION
**Pull request by lx75249**

### CODE REVIEW

**Comment by antinucleon** [comment]
Basically we are using ```grad_scale``` in optimizer to do this function, so I think in this view it won't be fixed.

**Comment by lx75249** [comment]
I assume mxnet uses cross entropy as the loss function of softmax, so mathematically its gradient should be divided by n. You can see https://github.com/dmlc/mxnet/blob/master/src/operator/regression_output-inl.h#L76, where the gradient is also divided by num_ouptut.

**Comment by antinucleon** [comment]
Thank you again for contribution! I will add a normalization Parma today,
which will also consider ignore label case. I am worried current change
will break many example and old code.
On Mon, May 23, 2016 at 11:58 Xin Li <notifications@github.com> wrote:

> I assume mxnet uses cross entropy as the loss function of softmax, so
> mathematically its gradient should be divided by n. You can see
> https://github.com/dmlc/mxnet/blob/master/src/operator/regression_output-inl.h#L76,
> where the gradient is also divided by num_ouptut.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/pull/2215#issuecomment-221058321>
>
-- 
Sent from mobile phone

**Comment by piiswrong** [comment]
You are confusing multi target with batch_size, operators normalize for the first while the second is done by optimizers

**Comment by antinucleon** [comment]
Close for now
